### PR TITLE
Add flake8 plugins

### PR DESCRIPTION
**Pull request by SharonHart**
Add additional plugins to flake8, used to be covered by pylint.
Moving to plugins, helps up control better what we're linting, and avoid exclusions. 
Also, plugin checks automatically added to flake8 checking in the CI, and to the pre-commit.

run `pipenv run flake8 --version` to see the effective plugins. 

More plugins can be found here:
https://github.com/DmytroLitvinov/awesome-flake8-extensions
Let me know if you see anything that we might need.

**Currently excluding analyzer from pre-commit and adding exclusion list since it's not aligned with flake8 (mainly docsting)**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,4 +8,10 @@ repos:
       rev: 3.7.9
       hooks:
         - id: flake8
+          additional_dependencies: [
+              'pep8-naming',
+              'flake8-docstrings',
+          ]
           args: ['--max-line-length=88']
+          exclude: ^presidio-analyzer/
```
Comment (SharonHart) [inline .pre-commit-config.yaml:16]: remove once analyzer will be aligned with flake8 (mainly docstrings)

File: presidio-analyzer/setup.cfg [hunk 2]
```diff
@@ -1,2 +1,4 @@
 [flake8]
 max-line-length = 88
+
+extend-ignore = D100,D101,D102,D104,D105,D107,D103,D200,D202,D205,D208,D210,E251,D400,D401,D403,F401,W503,N803
```
Comment (SharonHart) [inline presidio-analyzer/setup.cfg:4]: remove once analyzer will be aligned with flake8 (mainly docstrings)

File: presidio-analyzer/tests/test_analyzer_engine.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
-import os
```
Comment (SharonHart) [inline presidio-analyzer/tests/test_analyzer_engine.py:1]: black automformatting

File: presidio-analyzer/tests/test_iban_recognizer.py [hunk 4]
```diff
@@ -32,300 +32,860 @@ def update_iban_checksum(iban):
 @pytest.mark.parametrize(
     "iban, expected_len, expected_res",
     [
-        ("AL47212110090000000235698741", 1, ((0, 28),),),
-        ("AL47 2121 1009 0000 0002 3569 8741", 1, ((0, 34),),),
+        (
```
Comment (SharonHart) [inline presidio-analyzer/tests/test_iban_recognizer.py:35]: Actually this is less readable as a parametrized scenario, will see how to exclude this :(

File: presidio-analyzer/tests/test_aba_routing_recognizer.py [hunk 5]
```diff
@@ -17,6 +17,7 @@ def entities():
 @pytest.mark.parametrize(
     "text, expected_len, expected_positions, expected_score",
     [
+        # fmt: off
```
Comment (omri374) [inline presidio-analyzer/tests/test_aba_routing_recognizer.py:20]: Why are these non-compliant? just curious

**Review by omri374 — APPROVED** [review]

File: presidio-analyzer/tests/test_aba_routing_recognizer.py (hunk 5 above)
Comment (SharonHart) [reply presidio-analyzer/tests/test_aba_routing_recognizer.py:20]: @omri374 it is, black completely ruins readability
image
vs
image
Also take a look on the teams channel for more decisions / questions

File: presidio-analyzer/setup.cfg (hunk 2 above)
Comment (shiranr) [reply presidio-analyzer/setup.cfg:4]: Is there a user-story on this?

File: presidio-anonymizer/setup.cfg [hunk 6]
```diff
@@ -1,2 +1,5 @@
 [flake8]
-max-line-length = 88
\ No newline at end of file
+max-line-length = 88
+
+;remove requirement for init documentation and module
+extend-ignore = D100,D104,D107
```
Comment (SharonHart) [inline presidio-anonymizer/setup.cfg:5]: @shiranr why adding those?

File: presidio-analyzer/setup.cfg (hunk 2 above)
Comment (omri374) [reply presidio-analyzer/setup.cfg:4]: Yes

File: presidio-anonymizer/setup.cfg (hunk 6 above)
Comment (shiranr) [reply presidio-anonymizer/setup.cfg:5]: We need to discuss this in our daily/kickoff today. I think this should be a discussion.
IMHO there is such thing as too much comments, I don't believe in adding a comment so the linter will be happy.
A comment should be justified.

Comment (SharonHart) [reply presidio-anonymizer/setup.cfg:5]: Sure, let's investigate and talk, a random list here is not justified either...

File: VERSION [hunk 7]
```diff
@@ -1 +1 @@
-1.1.0
\ No newline at end of file
+1.2.0
```
Comment (shiranr) [inline VERSION:1]: I want to consult with @balteravishay regarding this... not sure if that's what we want :) by the time the project will be out we will be at version 1.100.0

File: .pre-commit-config.yaml [hunk 8]
```diff
@@ -8,4 +8,10 @@ repos:
       rev: 3.7.9
       hooks:
         - id: flake8
-          args: ['--max-line-length=88']
+          additional_dependencies: [
+              'pep8-naming',
+              'flake8-docstrings',
+          ]
+          args: ['--max-line-length=88', '--docstring-convention=numpy', '--per-file-ignores=**/tests/**.py:D']
+          exclude: ^presidio-analyzer/ #TODO: Remove once analyzer will be aligned with flake8 (mainly docstrings)
```
Comment (shiranr) [inline .pre-commit-config.yaml:16]: Please add user story number.

File: VERSION (hunk 7 above)
Comment (SharonHart) [reply VERSION:1]: We can keep the discussion in the teams

Comment (SharonHart) [reply VERSION:1]: @shiranr So minor it is, leaving it as is.

**Review by shiranr — APPROVED** [review]

**Review by omri374 — APPROVED** [review]

**Pull request merged**